frontend/packages/topology/src/components/page/TopologyView.tsx

```diff
@@ -320,25 +323,29 @@ export const ConnectedTopologyView: React.FC<ComponentProps> = ({
     }
   }, [searchParams, labelParams]);
 
+  const nodesLength = filteredModel?.nodes?.length ?? 0;
+
   const viewContent = React.useMemo(
     () =>
-      viewType === TopologyViewType.list ? (
-        <TopologyListView
-          model={filteredModel}
-          namespace={namespace}
-          onSelect={onSelect}
-          setVisualization={setVisualization}
-        />
-      ) : (
-        <Topology
-          model={filteredModel}
-          namespace={namespace}
-          application={applicationRef.current}
-          onSelect={onSelect}
-          setVisualization={setVisualization}
-        />
-      ),
-    [filteredModel, namespace, onSelect, viewType],
+      nodesLength < MAX_NODES_LIMIT || showTopologyAnyway ? (
```

Comment:
you use `nodesLength > MAX_NODES_LIMIT` when checking if the limit is exceeded, therefore you need to use `<=` here